lib/iris/fileformats/pp.py

```diff
@@ -1040,7 +1040,8 @@ def _data_bytes_to_shaped_array(data_bytes, lbpack, boundary_packing,
 
     # Mask the array?
     if mdi in data:
-        data = array_masked_to_nans(data, data == mdi)
+        data = np.ma.masked_array(data, data == mdi)
```

Comment:
@marqh we're seeing that there's no obvious answer to this. When I discussed this offline with @pp-mo he raised similar concerns about what was happening here...